src/nft/utils/calcPoolPrice.ts

```diff
@@ -1,11 +1,72 @@
 import { BigNumber } from '@ethersproject/bignumber'
 import { BagItem, BagItemStatus, GenieAsset, Markets, UpdatedGenieAsset } from 'nft/types'
 
+const PRECISION = '1000000000000000000'
+const PROTOCOL_FEE_MULTIPLIER = BigNumber.from('5000000000000000')
+
+enum BondingCurve {
+  Linear = 'LINEAR',
+  Exponential = 'EXPONENTIAL',
+}
+
+interface Pool {
+  delta?: string
+  spotPrice?: string
+  fee?: string
+  bondingCurve?: BondingCurve
+}
+
+const calcSudoSwapPoolPrice = (
```

Comment:
please say this was given to us by sudoswap and is also used in the backend

Reply:
that is true